File: backend/app/schemas/algorithm_package.py
import uuid
from datetime import datetime
from typing import Any

from pydantic import BaseModel


class AlgorithmPackageResponse(BaseModel):
    id: uuid.UUID
    name: str
    framework: str
    description: str | None
    status: str
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


class AlgorithmPackageVersionResponse(BaseModel):
    id: uuid.UUID
    package_id: uuid.UUID
    version: str
    entrypoint: str
    runtime_config: dict[str, Any] | None
    weights_path: str | None
    status: str
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}

File: frontend/src/types/algorithmPackage.ts
export interface AlgorithmPackage {
  id: string
  name: string
  framework: string
  description: string | null
  status: string
  created_at: string
  updated_at: string
}

export interface AlgorithmPackageVersion {
  id: string
  package_id: string
  version: string
  entrypoint: string
  runtime_config: Record<string, unknown> | null
  weights_path: string | null
  status: string
  created_at: string
  updated_at: string
}
